presto-main/src/main/java/com/facebook/presto/util/JsonUtil.java

```diff
@@ -527,4 +535,138 @@ public void writeJsonValue(JsonGenerator jsonGenerator, Block block, int positio
             }
         }
     }
+
+    // utility classes and functions for cast from JSON
+    public static Slice castTokenAsVarchar(JsonParser parser)
```

Comment:
As a verb, "parse" means:

```
analyze (a sentence) into its parts and describe their syntactic roles.
```

So looks like we cannot "parse" a token as varchar. We are casting a token as a varchar ... Token seems to be the output of an  intermediate stage of parsing (https://en.wikipedia.org/wiki/Parsing#Overview_of_process). :D